### ISSUE TITLE
Accessibility: Category filter navigation should be wrapped in nav landmark

### ISSUE THREAD

**Issue by philippsauer** [intershop/intershop-pwa#1834]
**Actual Behavior**

The category filter list, which contains links to individual category views, is currently rendered using generic `<div>` and `<ul>` elements. It is not wrapped within a `<nav>` landmark element, nor does it have an aria-label to identify its purpose.

As a result, assistive technologies like screen readers do not recognize this navigation block as a distinct landmark region. This hinders users from easily navigating to or understanding the purpose of the category filter area on the page.

**Expected Behavior**

The category filter list should be enclosed within a `<nav>` HTML5 semantic element. This element should be further enhanced with an aria-label (e.g., "Category Filter") to clearly describe its purpose to assistive technologies.

```
<nav aria-label="Category Filter">
   <div class="non-parametric-panel ng-star-inserted">
      ...
   </div>
</nav
```

**Steps to Reproduce the Bug**

1. Go to any category overview page (e.g: https://intershoppwa-b2b.azurewebsites.net/computers-ctgComputers).
2. Locate and Inspect the HTML structure of the category filter list in the sidebar
3. Observe that the `<ul class="filter-list">` and its parent `<div>` elements are not wrapped by a `<nav>` tag,

[AB#107463](https://dev.azure.com/intershop-com/cefd1005-00a7-4c79-927f-a16947d1b2e6/_workitems/edit/107463)
